### PR TITLE
Add Cloudflare R2 storage s3-compat profile

### PR DESCRIPTION
**Pull request by AliveDevil**
As part of #13376

Refer iterate-ch/docs#198

### CODE REVIEW

File: profiles/Cloudflare R2 Storage (S3).cyberduckprofile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        <string>Cloudflare R2 Storage (S3)</string>
+        <key>Default Hostname</key>
+        <string>accountid.r2.cloudflarestorage.com</string>
+        <key>Hostname Configurable</key>
+        <true/>
+        <key>Port Configurable</key>
+        <false/>
+        <key>Properties</key>
+        <array>
+            <string>s3service.disable-dns-buckets=true</string>
```
Comment (dkocher) [inline profiles/Cloudflare R2 Storage (S3).cyberduckprofile:21]: - Add setting for storage classes available
- Add definition for default region with `<key>Region</key>` and available regions `<key>Region</key>`
- Add help for https://docs.cyberduck.io/protocols/s3/cloudflare

Comment (dkocher) [inline profiles/Cloudflare R2 Storage (S3).cyberduckprofile:21]: - Add setting for storage classes available like `s3.storage.class.options=STANDARD GLACIER`
- Add definition for default region with `<key>Region</key>` and available regions `<key>Region</key>`
- Add help for https://docs.cyberduck.io/protocols/s3/cloudflare

Comment (AliveDevil) [reply profiles/Cloudflare R2 Storage (S3).cyberduckprofile:21]: Unimplemented:
> CreateMultipartUpload:
> ❌ Storage class:
> ❌ x-amz-storage-class

Didn't find any evidence that there are regions. Docs are opaque here.

Comment (dkocher) [reply profiles/Cloudflare R2 Storage (S3).cyberduckprofile:21]: Thus set 
- s3.storage.class.options=STANDARD
- `Region` and `Regions` explicitly to default to the value returned by the `?location` for buckets.

Comment (AliveDevil) [reply profiles/Cloudflare R2 Storage (S3).cyberduckprofile:23]: >  by the ?location for buckets
Awaiting response.

**Comment by dkocher** [comment]
Querying the bucket location is not supported

```
<Error><Code>NotImplemented</Code><Message>GetBucketLocation not implemented</Message></Error>
```

**Comment by AliveDevil** [comment]
RClone is using "auto" Region. Using same value.

File: profiles/Cloudflare R2 Storage (S3).cyberduckprofile (hunk 1 above)
Comment (AliveDevil) [reply profiles/Cloudflare R2 Storage (S3).cyberduckprofile:29]: Using "auto" value of rclone: https://forum.rclone.org/t/cloudflare-r2-now-working-with-rclone/30730

**Review by dkocher — APPROVED** [review]

**Comment by dkocher** [comment]
It may not be clear that `accountid` in the default hostname `accountid.r2.cloudflarestorage.com` needs to be replaced. I suppose it would be better to only set a placeholder now possible with #13396.

**Pull request merged**